auth/src/test/java/com/firebase/ui/auth/testhelpers/TestHelper.java

```diff
@@ -72,18 +74,42 @@ private static void spyContextAndResources() {
         when(RuntimeEnvironment.application.getResources()).thenReturn(spiedResources);
     }
 
-    private static FirebaseApp initializeApp(Context context) {
-        try {
-            return FirebaseApp.initializeApp(
-                    context,
-                    new FirebaseOptions.Builder()
-                            .setApiKey(API_KEY)
-                            .setApplicationId(APPLICATION_ID)
-                            .build(),
-                    FirebaseApp.DEFAULT_APP_NAME);
-        } catch (IllegalStateException e) {
-            return FirebaseApp.getInstance(FirebaseApp.DEFAULT_APP_NAME);
+    private static void initializeApp(Context context) {
+        if (!FirebaseApp.getApps(context).isEmpty()) { return; }
+
+        for (Field field : FirebaseApp.class.getDeclaredFields()) {
+            field.setAccessible(true);
+
+            Object o;
+            try {
+                o = field.get(null);
+            } catch (IllegalAccessException e) {
+                throw new IllegalStateException(e);
+            } catch (NullPointerException e) {
+                continue; // Instance field, move on
```

Comment:
Seems like one hack is still here ;-)

Reply:
Lol, yeah. The more accurate title would be `Remove testing hacks in production in favor of hacks in testing code`.